sabnzbd/filesystem.py

```diff
@@ -1369,6 +1384,50 @@ def pathbrowser(path: str, show_hidden: bool = False, show_files: bool = False)
     return file_list
 
 
+if sys.platform == "linux":
+
+    def write_at_offset(fd: int, data: memoryview, offset: int) -> int:
+        return os.pwrite(fd, data, offset)
+
+elif sys.platform == "darwin":
```

Comment:
This whole block of code seems very non-SAB-style.
I would have expected to use 1 function with `if sabnzbd.WIN32` inside.
Or if it's really performance drive, not using `sys.platform` but still using `sabnzbd.WIN32` etc.